app/models/user.rb

```diff
@@ -19,4 +21,42 @@ class User < ApplicationRecord
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: VALID_EMAIL_REGEX },
                     uniqueness: { case_sensitive: false }
+
+  def friends
+    friends = friendships.includes(:friend).where(confirmed: true).references(:users)
+    friends.map(&:friend)
+  end
+
+  def pending_friends
+    requests = friendships.includes(:friend).where(confirmed: false).references(:users)
+    requests.map(&:friend)
+  end
+
+  # Users who have requested to be friends
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  end
```

Comment:
**`OPTIONAL`**: Even it's well working. Please, could you think about rewriting it like explained in this example

```ruby
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```

Reply:
I'll apply all the optional suggestions later. Thanks